### PR TITLE
stop jobs that do not start within 30 days

### PR DESCRIPTION
**Pull request by bmc-msft**
If a job does not start within 30 days, stop the job and mark all of the tasks as `failed`.

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/jobs.py [hunk 1]
```diff
@@ -36,6 +40,34 @@ def search_expired(cls) -> List["Job"]:
...
+            (datetime.utcnow() - JOB_NEVER_STARTED_DURATION).isoformat()
+        )
+        for job in cls.search(
+            query={
+                "state": [JobState.enabled],
+            },
+            raw_unchecked_filter=time_filter,
+        ):
+            # if the job has started, leave it be
+            if job.end_time is not None:
```
Comment (chkeita) [inline src/api-service/__app__/onefuzzlib/jobs.py:55]: can this be added to the db query?

**Review by chkeita — APPROVED** [review]

File: src/api-service/__app__/onefuzzlib/jobs.py (hunk 1 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/jobs.py:55]: I got a failure when I tried in a fresh instance, as the field won't be created yet.

Comment (chkeita) [reply src/api-service/__app__/onefuzzlib/jobs.py:55]: I am ok to leave it as is but there is trick to check fields that might not exist we use it to [find outdated nodes](https://github.com/microsoft/onefuzz/blob/657af9722ceae3acc2e027f4181b3c62e96555fd/src/api-service/__app__/onefuzzlib/workers/nodes.py#L113)

Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/jobs.py:55]: oh neat.  I'll investigate that.

Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/jobs.py]: Updated following that example.

**Review by chkeita — APPROVED** [review]

**Pull request merged**